File: functions practice/updateBudget.c
#include <stdio.h>
float income, transportation, rent, utilities, groceries, expenses, savings, total;

float input(char type[], float vari){
    printf("How much is your %s: \n", type);
     scanf("%f", &vari);
     return vari;
}


void percent(char type[], int amount){
    int per = amount/income*100;
    printf("Your %s is %d%% of your income\n", type, per);
}


    int main(void){
    printf("This is going to calculate your budget for you\n");
    income = input("income", income);
    rent = input("rent", rent);
    utilities = input("utilities", utilities);
    groceries = input("groceries", groceries);
    transportation = input("transportation", transportation);
    expenses = rent + utilities + groceries + transportation;
    savings = income *.2;
    total = income - expenses - savings;

    printf("Your income is: $%.2f\n", income);
    printf("Your expenses is: $%.2f\n", expenses);
    printf("Your savings are: $%.2f\n", savings);
    printf("Your total left to spend is: $%.2f\n", total);

    percent("rent", rent);
    percent("utilities", utilities);
    percent("groceries", groceries);
    percent("transportation", transportation);
    percent("savings", savings);
    return 0;
}

File: functions practice/updateBudget.py

def dollars(type):

    return input(f"How much is your {type}:")

print("This is going to calculate your budget for you")
income = float(dollars("income"))
rent = float(dollars("rent"))
utilities = float(dollars("utilities"))
groceries = float(dollars("groceries"))
transportation = float(dollars("transportation"))

expenses = rent + utilities + groceries + transportation
savings = income * .2
total = income - expenses - savings
prent = (rent / income)* 100
putilities = (utilities / income)* 100
pgroceries = (groceries / income)* 100
ptransportation = (transportation / income)* 100
pexpenses = (expenses / income)* 100
psavings = (savings / income)* 100

def percent(type, amount):
    percentage = amount/income*100
    
    return(f"Your {type} is {int (percentage)}% of your income.")

print(f"Your income is: ${income:.2f}")
print(f"Your expenses are: ${expenses:.2f}")
print(f"Your savings are: ${savings:.2f}")
print(f"Your total left to spend is: ${total:.2f}")

print(percent("rent",(rent)))
print(percent("utilities", (utilities)))
print(percent("groceries", (groceries)))
print(percent("transportation", (transportation)))
print(percent("savings", (savings)))
print(percent("expenses", (expenses)))